### PR TITLE
Add HTTP/2 support

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public List<String> get(@Nullable String key)
+    {
+        return delegate.get(getKey(key));
+    }
+
+    @Override
+    public boolean containsKey(@Nullable Object key)
```
Comment (dain) [inline http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java]: There are only three methods that ned to be overwritten?  I would expect a bunch more.

**Comment by dain** [comment]
Looks good

File: http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java (hunk 1 above)
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java]: I think this is all that can be done with this interface. We should probably switch to a custom one.

Any ideas on other methods to override?

Comment (dain) [inline http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java]: The problem is all derived collections will be case sensitive.  For example, `asMap` and `keySet`

Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java]: Yeah, but I don't think we should deal with that here. It already violates the contract for a collection. This is probably good enough. But if you think we should add it, I can do that.

**Comment by dain** [comment]
looks good

**Pull request merged**